pkg/controller/infrastructure/infraflow/utils.go

```diff
@@ -34,6 +33,12 @@ func findExisting[T any](id *string, name string,
 	if len(found) == 0 {
 		return nil, nil
 	}
+
+	// TODO: check if this makes sense
```

Comment:
I guess this makes sense

Reply:
I guess so. The `findExisting` is geared towards finding a single result. I think it's better to return an error if we do find multiple instead of arbitrarily picking one of them (past behavior).